### PR TITLE
Fix global config get for port scan.

### LINKED ISSUE #2795: "cylc scan -a" broken on master
If no global.rc "suite servers" section exists:
```
$ cylc scan -a
Traceback (most recent call last):
  File "/home/oliverh/cylc/bin/cylc-scan", line 354, in <module>
    main()
  File "/home/oliverh/cylc/bin/cylc-scan", line 213, in main
    glbl_cfg().get(["suite servers", "scan hosts"], ["localhost"]))
  File "/home/oliverh/cylc/lib/parsec/config.py", line 113, in get
    raise ItemNotFoundError(itemstr(parents, key))
parsec.config.ItemNotFoundError: ERROR: item not found: suite servers
```

### PR DESCRIPTION
**Pull request by hjoliver**
Fix #2795 (bug on current master, since last release).

 `glbl_cfg().get()` isn't `dict.get()`.

(maybe was not the best name for that function).

Also removes a `cylc scan` test that was not actually testing anything anymore.

### CODE REVIEW

File: lib/cylc/mp_pool.py [hunk 1]
```diff
@@ -236,7 +236,7 @@ class SuiteProcPool(object):
 
     def __init__(self, size=None):
         if not size:
-            size = glbl_cfg().get(['process pool size'], size)
```
Comment (hjoliver) [inline lib/cylc/mp_pool.py]: Another instance of the same error (the only other one I found; and in this case it was of no consequence).

**Comment by matthewrmshin** [comment]
Note I took out the default settings specially. I think it is important to respect the empty setting, e.g. in the dumper and in logical decision in some cases - where the empty list should not be treated the same way as the default list.

(I also suspect that I've done something about default evaluation in #2609.)

**Comment by hjoliver** [comment]
Note we have the "--sparse" option to return explicitly set (i.e. non-default items).

I agree that some default settings should be empty or null.  But in this case, if we default to running suite daemons on localhost, is there any reason not to default to scanning localhost?  If not, then having to explicitly (and correctly) default to "localhost" at every use of the config item just seems like more chances to make errors, no?

**Comment by matthewrmshin** [comment]
(Not sure why we were both doing work on a Sunday!) The other problem is that we still need to handle the case when users put an empty value to the setting (so overriding the default). This still needs to be handled by the run time logic, so we might as well move the default to where the run time logic is.

**Comment by hjoliver** [comment]
No worries.  OK, saying we should default to "localhost" no matter what - even if the user explicit unsets the hosts list ... and that requires run time defaulting (because a built-in default gets overridden by the user setting).   I guess that makes sense, reverting to run time default...

**Comment by hjoliver** [comment]
(mind you, shouldn't that same arg apply to the list of ports, which you still set a built-in default for?)

**Comment by hjoliver** [comment]
(you win :+1: ... although I think we're not being consistent on this defaulting business.. Maybe #2609 helps, I'm *almost* on to that review again...)

**Comment by matthewrmshin** [comment]
Yes, the port setting has the same issue - although it is less likely that users will override it with empty settings. (Another problem I have just realised - on dumping out the configuration - integer ranges get expanded to all values, so we should find a way to tuck them back into a `START..END` string.)

**Comment by sadielbartholomew** [comment]
@hjoliver can you tag me as soon as this is at a stage when there is agreement on an approach & you want me to review the implementation (possibly that is now)? It's not always obvious from the context of the comments if this is the case. Then I will review it as a priority. Thanks :)

**Comment by hjoliver** [comment]
> ... - integer ranges get expanded to all values ...

Yeah, that's not nice.

**Comment by hjoliver** [comment]
@sadielbartholomew - you can go ahead and review this if Matt is happy with the defaulting behaviour now, which I think he is.  It's actually a trivial change, it'll take you about 5 mins max :grin:

**Comment by matthewrmshin** [comment]
(Running test in my local environment.)

**Comment by matthewrmshin** [comment]
(Tests pass in my environment.)

**Comment by matthewrmshin** [comment]
@sadielbartholomew please review 2.

**Comment by sadielbartholomew** [comment]
Sorry for the delay. I have just run the full test battery locally with ``suite servers`` not defined (it passed, hooray). Just need to ensure it still works with it defined, so just awaiting the test results.

**Pull request merged**